sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -0,0 +1,503 @@
+        /// to "https://{registry-name}.azurecr.io".</param>
+        /// <param name="credential">The API key credential used to authenticate requests
+        /// against the container registry.  </param>
+        /// <param name="repository"> The name of the repository that logically groups the artifact parts</param>
+        /// <param name="options">Client configuration options for connecting to Azure Container Registry.</param>
+        /// <exception cref="ArgumentNullException"> Thrown when the <paramref name="endpoint"/>, <paramref name="credential"/>, or <paramref name="repository"/> is null. </exception>
+        public ContainerRegistryBlobClient(Uri endpoint, TokenCredential credential, string repository, ContainerRegistryClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+            Argument.AssertNotNull(repository, nameof(repository));
+
+            _endpoint = endpoint;
+            _registryName = endpoint.Host.Split('.')[0];
+            _repositoryName = repository;
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            _acrAuthPipeline = HttpPipelineBuilder.Build(options);
+            _acrAuthClient = new AuthenticationRestClient(_clientDiagnostics, _acrAuthPipeline, endpoint.AbsoluteUri);
+
+            string defaultScope = options.Audience + "/.default";
+            _pipeline = HttpPipelineBuilder.Build(options, new ContainerRegistryChallengeAuthenticationPolicy(credential, defaultScope, _acrAuthClient));
+            _restClient = new ContainerRegistryRestClient(_clientDiagnostics, _pipeline, _endpoint.AbsoluteUri);
+            _blobRestClient = new ContainerRegistryBlobRestClient(_clientDiagnostics, _pipeline, _endpoint.AbsoluteUri);
+        }
+
+        /// <summary> Initializes a new instance of ContainerRegistryBlobClient for mocking. </summary>
+        protected ContainerRegistryBlobClient()
+        {
+        }
+
+        /// <summary>
+        /// Gets the registry service endpoint for this client.
+        /// </summary>
+        public virtual Uri Endpoint => _endpoint;
+
+        /// <summary>
+        /// Gets the name of the repository that logically groups the artifact parts.
+        /// </summary>
+        public virtual string RepositoryName => _repositoryName;
+
+        /// <summary>
+        /// Upload an artifact manifest from a stream.
+        /// </summary>
+        /// <param name="stream">The stream containing the serialized manifest.</param>
+        /// <param name="options">Options for configuring the upload operation.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public virtual Response<UploadManifestResult> UploadManifest(Stream stream, UploadManifestOptions options = default, CancellationToken cancellationToken = default)
```

Comment:
Do we need a `DiagnosticScope` here?

Reply:
I think that's handled by the inner method... unless you know of an example where we do it in both places?

Reply:
You shouldn't do it in both places - there is validation that will prevent it.